components/provisioner/internal/provisioning/service.go

```diff
@@ -223,6 +229,47 @@ func (r *service) UpgradeGardenerShoot(runtimeID string, input gqlschema.Upgrade
 	return r.graphQLConverter.OperationStatusToGQLOperationStatus(operation), nil
 }
 
+func (r *service) HibernateCluster(runtimeID string) (*gqlschema.OperationStatus, apperrors.AppError) {
+	log.Infof("Starting hibernation for Runtime '%s'...", runtimeID)
+
+	session := r.dbSessionFactory.NewReadSession()
+
+	err := r.verifyLastOperationFinished(session, runtimeID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Maybe we can check here if the cluster is already hibernated and in such a case just skip the operation

Reply:
As we agreed it is not needed as the current implementation works very fast.